junit/kubernetes-server-mock/src/test/java/io/fabric8/kubernetes/client/server/mock/crud/KubernetesCrudDispatcherPostTest.java

```diff
@@ -178,7 +178,8 @@ void postNamespaceMismatch() throws Exception {
     final HttpResponse<String> result = httpClient.sendAsync(request, String.class).get(10, TimeUnit.SECONDS);
     // Then
     assertThat(result)
-        .hasFieldOrPropertyWithValue("response.code", 400)
+        .extracting(HttpResponse::code).isEqualTo(400);
+    assertThat(result)
```

Comment:
did you change this because it failed in Java17? or is there any other reason?

Reply:
a single assertion can be preserved using .returns(400, HttpResponse::code)

Reply:
This was needed due to the response wrapping - with some additional refactoring that could be avoided.

Reply:
OK, then I'll replace with the returns DSL method that should work